src/decoder/lattice-incremental-decoder.cc

```diff
@@ -1130,7 +1149,9 @@ bool LatticeIncrementalDecoderTpl<FST, Token>::GetIncrementalRawLattice(
       // For now, we use extra_cost from the decoding stage , which has some
       // "future information", as
       // the final weights of this chunk
-      BaseFloat cost_offset = tok->extra_cost;
+      // TODO: using extra_cost does not help (worse). The reason could be 
```

Comment:
this is the place using extra_cost in the end of a chunk

Reply:
I think maybe I told you wrong when I said use the extra_cost as the final-arc cost.  I think it really should have been extra_cost *minus* alpha.  Can you please try that?

Reply:
Ya it works